mcs/class/System.Net.Http/MonoWebRequestHandler.cs

```diff
@@ -446,6 +447,9 @@ public async Task<HttpResponseMessage> SendAsync (HttpRequestMessage request, Ca
 			var wrequest = CreateWebRequest (request);
 			HttpWebResponse wresponse = null;
 
+			if (timeout != null)
```

Comment:
All setup calls are in CreateWebRequest I'd move it there too

Reply:
Sure, good idea.